### PR TITLE
Fix parquet_write_test.py::test_hive_timestamp_value failure for Databricks [databricks]

### LINKED ISSUE #8726: [BUG] Test "parquet_write_test.py::test_hive_timestamp_value[INJECT_OOM]" failed on Databricks
**Describe the bug**
Test "parquet_write_test.py::test_hive_timestamp_value[INJECT_OOM]" failed on Databricks

Note: nightly build/test PASS out of Databricks env, like on Dataproc cluster.

```
 =================================== FAILURES ===================================
 __________________________ test_hive_timestamp_value ___________________________
 [gw1] linux -- Python 3.8.10 /usr/bin/python
 
 spark_tmp_table_factory = <conftest.TmpTableFactory object at 0x7f6f62ab1bb0>
 spark_tmp_path = '/tmp/pyspark_tests//0716-051143-4t97fgdc-10-59-174-113-gw1-2906-1709110285/'
     def test_hive_timestamp_value(spark_tmp_table_factory, spark_tmp_path):
         def create_table(spark, path):
             tmp_table = spark_tmp_table_factory.get()
             spark.sql(f"CREATE TABLE {tmp_table} STORED AS PARQUET " +
             f""" LOCATION '{path}' AS SELECT CAST('2015-01-01 00:00:00' AS TIMESTAMP) as t; """)
         def read_table(spark, path):
             return spark.read.parquet(path)
         data_path = spark_tmp_path + '/PARQUET_DATA'
 >       assert_gpu_and_cpu_writes_are_equal_collect(create_table, read_table, data_path)
 
 ../../src/main/python/parquet_write_test.py:745: 
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
 ../../src/main/python/asserts.py:267: in assert_gpu_and_cpu_writes_are_equal_collect
     _assert_gpu_and_cpu_writes_are_equal(write_func, read_func, base_path, 'COLLECT', conf=conf)
 ../../src/main/python/asserts.py:242: in _assert_gpu_and_cpu_writes_are_equal
     with_gpu_session(lambda spark : write_func(spark, gpu_path), conf=conf)
 ../../src/main/python/spark_session.py:133: in with_gpu_session
     return with_spark_session(func, conf=copy)
 ../../src/main/python/spark_session.py:100: in with_spark_session
     ret = func(_spark)
 ../../src/main/python/asserts.py:242: in <lambda>
     with_gpu_session(lambda spark : write_func(spark, gpu_path), conf=conf)
../../src/main/python/parquet_write_test.py:738: in create_table
    spark.sql(f"CREATE TABLE {tmp_table} STORED AS PARQUET " +
/databricks/spark/python/pyspark/instrumentation_utils.py:48: in wrapper
    res = func(*args, **kwargs)
/databricks/spark/python/pyspark/sql/session.py:1387: in sql
    return DataFrame(self._jsparkSession.sql(sqlQuery, litArgs), self)
/databricks/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py:1321: in __call__
    return_value = get_return_value(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = ('xro892112', <py4j.clientserver.JavaClient object at 0x7f6f99bf5f70>, 'o64', 'sql')
kw = {}, converted = IllegalArgumentException()

    def deco(*a: Any, **kw: Any) -> Any:
        try:
            return f(*a, **kw)
        except Py4JJavaError as e:
            converted = convert_exception(e.java_exception)
            if not isinstance(converted, UnknownException):
                # Hide where the exception came from that shows a non-Pythonic
                # JVM exception message.
>               raise converted from None
E               pyspark.errors.exceptions.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.command.DataWritingCommandExec
E               Execute InsertIntoHadoopFsRelationCommand file:/tmp/pyspark_tests/0716-051143-4t97fgdc-10-59-174-113-gw1-2906-1709110285/PARQUET_DATA/GPU, false, Parquet, [serialization.format=1, mergeschema=false, __hive_compatible_bucketed_table_insertion__=true], Append, `spark_catalog`.`default`.`tmp_table_gw1_2097635381_1`, org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe, org.apache.spark.sql.execution.datasources.InMemoryFileIndex(file:/tmp/pyspark_tests/0716-051143-4t97fgdc-10-59-174-113-gw1-2906-1709110285/PARQUET_DATA/GPU), [t]
E               +- WriteFiles
E                  +- GpuColumnarToRow false
E                     +- GpuProject [1420070400000000 AS t#1615897]
E                        +- GpuRowToColumnar targetsize(104857600)
E                           +- Scan OneRowRelation[]

/databricks/spark/python/pyspark/errors/exceptions.py:234: IllegalArgumentException
----------------------------- Captured stdout call -----------------------------
### CPU RUN ###
### GPU RUN ###
=============================== warnings summary ===============================
../../../../../../databricks/spark/python/pyspark/sql/pandas/utils.py:37: 16 warnings
src/main/python/udf_test.py: 430 warnings
  /databricks/spark/python/pyspark/sql/pandas/utils.py:37: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if LooseVersion(pandas.__version__) < LooseVersion(minimum_pandas_version):

../../../../../../databricks/spark/python/pyspark/sql/pandas/utils.py:64: 16 warnings
src/main/python/udf_test.py: 296 warnings
  /databricks/spark/python/pyspark/sql/pandas/utils.py:64: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if LooseVersion(pyarrow.__version__) < LooseVersion(minimum_pyarrow_version):

src/main/python/hash_aggregate_test.py: 32 warnings
  /databricks/spark/python/pyspark/sql/functions.py:723: FutureWarning: Deprecated in 3.2, use sum_distinct instead.
    warnings.warn("Deprecated in 3.2, use sum_distinct instead.", FutureWarning)

src/main/python/hive_delimited_text_test.py: 56 warnings
  /databricks/spark/python/pyspark/sql/catalog.py:763: FutureWarning: createExternalTable is deprecated since Spark 2.2, please use createTable instead.
    warnings.warn(

src/main/python/window_function_test.py::test_lru_cache_datagen[INJECT_OOM]
  /home/ubuntu/spark-rapids/integration_tests/src/main/python/window_function_test.py:1452: UserWarning: Cache info: CacheInfo(hits=11380, misses=971, maxsize=128, currsize=128)
    warnings.warn("Cache info: {}".format(info))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
- generated xml file: /home/ubuntu/spark-rapids/integration_tests/target/run_dir-20230716052256-1MOT/TEST-pytest-1689484976783667901.xml -
=========================== short test summary info ============================
FAILED ../../src/main/python/parquet_write_test.py::test_hive_timestamp_value[INJECT_OOM] - pyspark.errors.exceptions.IllegalArgumentException: Part of the plan is not...
= 1 failed, 18123 passed, 845 skipped, 848 xfailed, 291 xpassed, 847 warnings in 7336.30s (2:02:16) =
 Warning: Permanently added '[34.216.160.175]:2200' (ECDSA) to the list of known hosts.
 receiving incremental file list

```

### PR DESCRIPTION
**Pull request by razajafri**
Added `allow_non_gpu_databricks` to the test to pass for DB

fixes #8726

### CODE REVIEW

**Comment by jlowe** [comment]
Can you elaborate on why it's OK on Databricks for the write to fallback to the CPU?  By allowing the write to fallback to the CPU, it seems we're defeating half of the point of this test, as we're testing CPU writes against CPU writes.

**Comment by razajafri** [comment]
> Can you elaborate on why it's OK on Databricks for the write to fallback to the CPU? By allowing the write to fallback to the CPU, it seems we're defeating half of the point of this test, as we're testing CPU writes against CPU writes.

We don't write `LEGACY` rebase mode for timestamps on GPU on Databricks. So either I make a change where I only test the rebase mode to be `CORRECTED` or I just fallback on CPU for `LEGACY` on DB. The third option is probably the best which is to do both but will require to fallback on CPU for `LEGACY` for DB only. Right now we don't have an `allow_non_gpu_if` but I think it might be useful to have. 

I thought the first option was OK but if you strongly think I can change it to second or even the third option

File: integration_tests/src/main/python/parquet_write_test.py [hunk 1]
```diff
@@ -730,20 +730,44 @@ def write_partitions(spark, table_path):
...
+    hive_timestamp_value(spark_tmp_table_factory, spark_tmp_path, 'CORRECTED', func_test)
+
+# Test to avoid regression on a known bug in Spark. For details please visit https://github.com/NVIDIA/spark-rapids/issues/8693
+@allow_non_gpu('DataWritingCommandExec')
+def test_hive_timestamp_value_fallback(spark_tmp_table_factory, spark_tmp_path):
+
+    def func_test(create_table, read_table, data_path, conf):
+        assert_gpu_fallback_write(
+            lambda spark, path: create_table(spark, path),
+            lambda spark, path: read_table(spark, path),
```
Comment (jlowe) [inline integration_tests/src/main/python/parquet_write_test.py:765]:
```suggestion
            create_table,
            read_table,
```

**Review by jlowe — DISMISSED** [review]
Minor suggestion to remove an extraneous layer of indirection, lgtm.

**Comment by razajafri** [comment]
build

**Review by jlowe — APPROVED** [review]

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
CI failed due to the second issue mentioned in https://github.com/NVIDIA/spark-rapids/issues/8716. Looks like an intermittent issue. Kicking off CI again to unblock the DB nightly build

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
retriggered after https://github.com/NVIDIA/spark-rapids/pull/8747

**Review by jlowe — APPROVED** [review]

**Pull request merged**